File: 033. Cómo solucionar los problemas de fetching con Spring Data JPA/spring-data-jpa-fetching/src/main/java/com/example/repository/BookRepository.java
package com.example.repository;

import com.example.model.Book;
import com.example.projection.BookTitle;
import com.example.projection.BookView;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.List;

public interface BookRepository extends JpaRepository<Book, Long> {

    /*
    Otro enfoque sería utilizar la API Entity Graph
     */
    @Query("SELECT b FROM Book b join fetch b.author a")
    List<Book> findAllWithAuthor();

    @Query("SELECT b FROM Book b join fetch b.author a WHERE a.name = :name")
    List<Book> findAllWithAuthorByName(String name);

    @Query("SELECT b FROM Book b join fetch b.author a join fetch a.address ad")
    List<Book> findAllWithAuthorAndAddress();

    List<BookTitle> findTitleByAuthorName(String name);

    @Query("""
            SELECT
            b.title as bookTitle,
            a.name as authorName,
            ad.city as authorCity
            FROM Book b
            JOIN b.author a
            JOIN a.address ad
            WHERE b.title like :title
            """)
    List<BookView> findViewByTitleLike(String title);


}

File: 088. Implementa una API Reactiva con Spring WebFlux y R2DBC con Kotlin/spring-webflux-r2dbc-kotlin/src/main/kotlin/com/example/repository/BookRepository.kt
package com.example.repository

import com.example.domain.Book
import org.springframework.data.r2dbc.repository.Query
import org.springframework.data.r2dbc.repository.R2dbcRepository
import reactor.core.publisher.Flux

interface BookRepository : R2dbcRepository<Book, Long> {

    @Query("SELECT * FROM books WHERE author = :author")
    fun findAllByAuthor(author: String): Flux<Book>
}